### PR TITLE
Feat: support port => with field reference

### PR DESCRIPTION
**Pull request by kares**
We're extending the required `port => ...` option with support for dynamic field (target port) values.
If the target port does not respond events are dropped.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 3.2.0
+ - Feat: support port => with field reference [#13](https://github.com/logstash-plugins/logstash-output-udp/pull/13)
```
Comment (andsel) [inline CHANGELOG.md:2]:
```suggestion
 - Added support for field reference to `port`options [#13](https://github.com/logstash-plugins/logstash-output-udp/pull/13)
```

File: lib/logstash/outputs/udp.rb [hunk 2]
```diff
@@ -27,7 +28,14 @@ def register
     @socket = UDPSocket.new
 
     @codec.on_event do |event, payload|
-      socket_send(payload)
+      port = event.sprintf(@port)
+      begin
+        port = Integer(port)
+      rescue => e # ArgumentError (invalid value for Integer(): "foo")
+        logger.error("Failed to resolve port, dropping event", port: @port, event: event.to_hash, message: e.message)
```
Comment (andsel) [inline lib/logstash/outputs/udp.rb:35]: Did you evaluate to put the failing event into DLQ?

**Review by andsel — APPROVED** [review]
LGTM
I've left a comment to improve the changelog string and a question on DLQ. Maybe DLQ could be an overkill for this PR and maybe follow up in another one.

File: lib/logstash/outputs/udp.rb (hunk 2 above)
Comment (kares) [reply lib/logstash/outputs/udp.rb:35]: Not really, but did have a chat about the feature and dropping was considered acceptable.

**Pull request merged**